tests/test_web/test_healthcheck.py

```diff
@@ -0,0 +1,240 @@
+
+
+@pytest.mark.parametrize('short', [True, False])
+@mock.patch('iib.web.healthcheck.get_worker_config')
+@mock.patch('iib.web.healthcheck.database_connection_check')
+@mock.patch('iib.web.healthcheck.rabbitmq_connection_check')
+@mock.patch('iib.web.healthcheck.workers_status')
+def test_status_database_not_okay(
+    mock_workers_status,
+    mock_rabbitmq_connection_check,
+    mock_database_connection_check,
+    mock_get_worker_config,
+    short,
+):
+
+    config = mock_worker_config()
+
+    mock_get_worker_config.return_value = config
+
+    mock_database_connection_check.return_value = (False, 'database connection failed')
+    mock_rabbitmq_connection_check.return_value = (True, None)
+    mock_workers_status.return_value = [{'name': 'celery@a868abda4f69', 'available': True}]
+
+    if short:
+        with pytest.raises(IIBError, match='DATABASE unavailable: database connection failed'):
+            healthcheck.status(short=True)
+        return
+
+    result = healthcheck.status(short=False)
+
+    expected_services = [
+        {'name': 'DATABASE', 'available': False, 'reason': 'database connection failed'},
+        {'name': 'RABBITMQ', 'available': True},
+    ]
+
+    assert result == {
+        'services': expected_services,
+        'workers': mock_workers_status.return_value,
+    }
+
+    mock_database_connection_check.assert_called_once()
+    mock_rabbitmq_connection_check.assert_called_once_with(config.broker_url)
+    mock_workers_status.assert_called_once_with(retries=2)
+
+
+@pytest.mark.parametrize('short', [True, False])
+@pytest.mark.parametrize(
+    'workers_status, expected_result',
+    [
+        ([], False),
+        ([{'name': 'celery@a868abda4f68', 'available': False, 'reason': 'unknown error'}], False),
+        (
+            [
+                {'name': 'celery@a868abda4f68', 'available': False, 'reason': 'unknown error'},
+                {'name': 'celery@a868abda4f69', 'available': True},
+            ],
+            True,
+        ),
+    ],
+)
+@mock.patch('iib.web.healthcheck.get_worker_config')
+@mock.patch('iib.web.healthcheck.database_connection_check')
+@mock.patch('iib.web.healthcheck.rabbitmq_connection_check')
+@mock.patch('iib.web.healthcheck.workers_status')
+def test_status_workers_not_okay(
+    mock_workers_status,
+    mock_rabbitmq_connection_check,
+    mock_database_connection_check,
+    mock_get_worker_config,
+    short,
+    workers_status,
+    expected_result,
+):
+
+    config = mock_worker_config()
+
+    mock_get_worker_config.return_value = config
+
+    mock_database_connection_check.return_value = (True, None)
+    mock_rabbitmq_connection_check.return_value = (True, None)
+    mock_workers_status.return_value = workers_status
+
+    if short and not expected_result:
+        with pytest.raises(IIBError, match='no workers are available'):
+            healthcheck.status(short=True)
+        return
+
+    result = healthcheck.status(short=False)
```

Comment:
should not this `(short=False)` be `(short=short)` instead? Or why do you have the short parametrized?

Reply:
Can you explain to me why do you think it should be `short=short` and not `short=False` ?

Reply:
Please, do not resolve conversation opened by others. I mean why not to use the variable from parametrize?

Reply:
I've always done it this way, resolving after addressing the reviewer's concerns. I marked it as resolved as there is no change required here. If you see there is another approach, that's great, but I don't think I'll change these tests :)

Reply:
@zanssa I agree, we should use the variable value instead of hard coded values here since you have parameterized the value already. 

It's very easy for someone else altering your test to miss the hard coded value. +1 on using the variable name.